contracts/drafts/ERC777/ERC777.sol

```diff
@@ -0,0 +1,402 @@
+        require(to != address(0));
+        require((amount % _granularity) == 0);
+
+        // revert if 'to' is a contract not implementing tokensReceived()
+        require(_callTokensReceived(operator, address(0), to, amount, userData, operatorData));
+
+        // Update state variables
+        _totalSupply = _totalSupply.add(amount);
+        _balances[to] = _balances[to].add(amount);
+        assert((_balances[to] % _granularity) == 0);
+
+        emit Minted(operator, to, amount, userData, operatorData);
+    }
+
+    /**
+    * @dev Authorize an operator for the sender
+    * @param operator address to be authorized as operator
+     */
+    function _authorizeOperator(address operator) private {
+        _operators[msg.sender][operator] = true;
+        emit AuthorizedOperator(operator, msg.sender);
+    }
+
+    /**
+    * @dev Re-authorize a previously revoked default operator
+    * @param operator address to be re-authorized as operator
+     */
+    function _reAuthorizeDefaultOperator(address operator) private {
+        delete _revokedDefaultOperators[msg.sender][operator];
+        emit AuthorizedOperator(operator, msg.sender);
+    }
+
+    /**
+    * @dev Revoke operator rights from one of the default operators
+    * @param operator address to revoke operator rights from
+    */
+    function _revokeDefaultOperator(address operator) private {
+        _revokedDefaultOperators[msg.sender][operator] = true;
+        emit RevokedOperator(operator, msg.sender);
+    }
+
+    /**
+    * @dev Revoke an operator for the sender
+    * @param operator address to revoke operator rights from
+    */
+    function _revokeOperator(address operator) private {
+        delete _operators[msg.sender][operator];
+        emit RevokedOperator(operator, msg.sender);
+    }
+
+    /**
+     * @dev Send tokens
+     * @param operator address operator requesting the transfer
+     * @param from address token holder address
+     * @param to address recipient address
+     * @param amount uint256 amount of tokens to transfer
+     * @param userData bytes extra information provided by the token holder (if any)
+     * @param operatorData bytes extra information provided by the operator (if any)
+     */
+    function _send(
+        address operator,
+        address from,
+        address to,
+        uint256 amount,
+        bytes memory userData,
+        bytes memory operatorData
+    )
+    private
+    {
+        require(from != address(0));
+        require(to != address(0));
+        require((amount % _granularity) == 0);
+
+        _callTokensToSend(operator, from, to, amount, userData, operatorData);
+
+        // Update state variables
+        _balances[from] = _balances[from].sub(amount);
+        _balances[to] = _balances[to].add(amount);
+        assert((_balances[from] % _granularity) == 0);
+        assert((_balances[to] % _granularity) == 0);
+
+        _callTokensReceived(operator, from, to, amount, userData, operatorData);
```

Comment:
`_callTokensReceived` returns a boolean value that is being ignored. :upside_down_face:

We should remove that return value and make the function revert itself, just like `_callTokensToSend`.